File: 1547.js
const fs = require('fs');
const data = fs.readFileSync(0, 'utf8').trim().split(/\s+/).map(Number);

let idx = 0;
let n = data[idx++];

for(let i = 0; i < n; i++) {
    let QT = data[idx++];
    let S = data[idx++];

    let menor = 1000;
    let pos = 0;
    for(let j = 0; j < QT; j++) {
        let quess = data[idx++];
        let diff = Math.abs(S - quess);
        if(diff < menor) {
            menor = diff;
            pos = j + 1;
        }
    }

    console.log(pos);
}

File: 1547.c
#include <math.h>
#include <stdio.h>

int main(void){
    int n, i, j, QT, S, quess, pos, menor;

    scanf("%d", &n);
    for(i = 0; i < n; i++) {
        scanf("%d %d", &QT, &S);

        pos = 0;
        menor = 1000;
        for(j = 0; j < QT;  j++) {
            scanf("%d", &quess);

            if(abs(S - quess) < menor) {
                menor = abs(S - quess);
                pos = j + 1;
            }
        }

        printf("%d\n", pos);
    }


    return 0;
}